internal/service/employeeaccessgrant/resource.go

```diff
@@ -29,19 +36,54 @@ func (r *employeeAccessGrantRS) Schema(ctx context.Context, req resource.SchemaR
 }
 
 func (r *employeeAccessGrantRS) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
-	var tfPlan TFEmployeeAccessGrantModel
-	resp.Diagnostics.Append(req.Plan.Get(ctx, &tfPlan)...)
+	var tfModel TFModel
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &tfModel)...)
 	if resp.Diagnostics.HasError() {
 		return
 	}
-	resp.Diagnostics.Append(resp.State.Set(ctx, tfPlan)...)
+	atlasReq, err := NewAtlasReq(&tfModel)
+	if err != nil {
+		resp.Diagnostics.AddError(errorCreate, err.Error())
+		return
+	}
+	connV2 := r.Client.AtlasV2
+	projectID := tfModel.ProjectID.ValueString()
+	clusterName := tfModel.ClusterName.ValueString()
+	if _, _, err := connV2.ClustersApi.GrantMongoDBEmployeeAccess(ctx, projectID, clusterName, atlasReq).Execute(); err != nil {
+		resp.Diagnostics.AddError(errorCreate, err.Error())
+		return
+	}
+	resp.Diagnostics.Append(resp.State.Set(ctx, tfModel)...)
 }
 
 func (r *employeeAccessGrantRS) Read(ctx context.Context, req resource.ReadRequest, resp *resource.ReadResponse) {
+	var tfModel TFModel
+	resp.Diagnostics.Append(req.State.Get(ctx, &tfModel)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+	connV2 := r.Client.AtlasV2
+	projectID := tfModel.ProjectID.ValueString()
+	clusterName := tfModel.ClusterName.ValueString()
+	cluster, httpResp, err := connV2.ClustersApi.GetCluster(ctx, projectID, clusterName).Execute()
+	if httpResp.StatusCode == http.StatusNotFound {
+		resp.State.RemoveResource(ctx)
+		return
+	}
+	if err != nil {
+		resp.Diagnostics.AddError(errorCreate, err.Error())
+		return
+	}
+	atlasResp, _ := cluster.GetMongoDBEmployeeAccessGrantOk()
+	if atlasResp == nil {
+		resp.State.RemoveResource(ctx)
+		return
+	}
```

Comment:
so once a grant expires, the resource would be marked as removed right? believe this is the correct outcome but would make sure to call out in documentation

Reply:
Not exactly,  this is in case the grant is revoked outside from TF, e.g. UI or directly calling the API. 
The first not found is in case the cluster was deleted (so no grants associated any more).

Expiration grants are not deleted automatically.